service_linux.go

```diff
@@ -185,6 +182,15 @@ func (c *controller) rmServiceBinding(name, sid, nid, eid string, vip net.IP, in
 	if len(vip) != 0 {
 		n.(*network).rmLBBackend(ip, vip, lb.fwMark, ingressPorts, rmService)
 	}
+	s.Unlock()
```

Comment:
Question: Is it required for the service lock to be held while we remove the backend ?
Just looking at the code, it seems it could have been released earlier after checking `if len(s.loadBalancers) == 0 {`